pinot-transport/src/main/java/com/linkedin/pinot/routing/HelixExternalViewBasedRouting.java

```diff
@@ -65,10 +67,15 @@
  */
 public class HelixExternalViewBasedRouting implements RoutingTable {
   private static final Logger LOGGER = LoggerFactory.getLogger(HelixExternalViewBasedRouting.class);
-  private final RoutingTableBuilder _offlineRoutingTableBuilder;
+
+  private final RoutingTableBuilder _largeClusterRoutingTableBuilder;
+  private final RoutingTableBuilder _smallClusterRoutingTableBuilder;
   private final RoutingTableBuilder _realtimeHLCRoutingTableBuilder;
   private final RoutingTableBuilder _realtimeLLCRoutingTableBuilder;
 
+  private static final int MIN_SERVER_COUNT_FOR_LARGE_CLUSTER = 30;
+  private static final int MIN_REPLICA_COUNT_FOR_LARGE_CLUSTER = 4;
```

Comment:
Talked to JF. He said that most high qps cases have larger number of replicas to handle the qps.